pkg/agent/datamodel/sig_config.go

```diff
@@ -12,6 +15,30 @@ const (
 	AzurePublicCloudSigSubscription string = "109a5e88-712a-48ae-9078-9ca8b3c81345" // AKS VHD
 )
 
+//nolint:gochecknoglobals
+var (
+	CachedFromComponents = make(map[string]ProcessedComponents)
+	CachedFromManifest   = make(map[string]ProcessedManifest)
+)
+
+//nolint:gochecknoinits
+func init() {
+	CacheManifest()
+	CacheComponents()
+}
+
+func CacheManifest() {
+	_, filename, _, _ := runtime.Caller(0)
+	manifestFilePath := "../../../parts/linux/cloud-init/artifacts/manifest.json"
```

Comment:
can't we just read from the relative path directly?

Reply:
hoping @element-of-surprise's change can make it in so we could just take from the embedded FS instead...

Reply:
We cannot add the relative path, when you run a go test, it assumes you are at whatever directory you are running 'go test' which causes problems where it thinkings the path is empty. Also we did explore the embedded path, but since the manifest.json and components.json are not in the same directory, it didn't work. Also manifest has the #EOF at the end that makes it tricky also.

Reply:
yeah that was using raw embedding, though John's change would have it such that we can access all embedded files essentially as its own filesystem, where all paths are accessible - though as I mentioned that's not available as of yet, so this is fine for now